File: app/Http/Controllers/SpecialistPatternsJava/Creational/AbstractFactory/AbstractFactoryJava/src/walls/WoodWall.java
package walls;

// Класс WoodWall представляет стену из дерева.
public class WoodWall implements Wall {

    // Метод build() выполняет построение стены из дерева.
    @Override
    public void build() {
        System.out.println("Постройка стены из дерева");
    }
}

File: app/Http/Controllers/SpecialistPatterns/AbstractFactory/Walls/WoodWall.php
<?php

namespace App\Http\Controllers\SpecialistPatterns\AbstractFactory\Walls;

// Класс WoodWall представляет стену из дерева.
use Illuminate\Support\Facades\Log;

class WoodWall implements Wall {

    // Метод build() выполняет построение стены из дерева.
    public function build() {
        Log::channel('sergey')->info("Постройка стены из дерева");
    }
}
